### PR TITLE
Add podcast by feed url

### PR DESCRIPTION
**Pull request by mitchdowney**

### CODE REVIEW

File: src/state/actions/downloads.ts [hunk 1]
```diff
@@ -47,19 +47,23 @@ export const initDownloads = async () => {
...
-    downloadsActive,
-    downloadsArray,
-    downloadedEpisodeIds,
-    downloadedEpisodeLimitCount,
-    downloadedEpisodeLimitDefault,
-    downloadedPodcastEpisodeCounts,
-    downloadedPodcasts
-  })
+  // TODO: There is a race condition preventing this state from being set properly on app launch :(
+  // I don't know where the problem is coming from...
```
Comment (kreonjr) [inline src/state/actions/downloads.ts]: setGlobal is Asynchronous by nature. What values are you looking for and where that you expected them to be populated in time?

File: src/screens/PodcastScreen.tsx [hunk 2]
```diff
@@ -95,15 +97,20 @@ export class PodcastScreen extends React.Component<Props, State> {
...
         <RNView style={core.row}>
-          <NavShareIcon
-            endingText=" – shared using Podverse"
-            podcastTitle={podcastTitle}
-            url={PV.URLs.podcast + podcastId}
-          />
+          {
+            !addByFeedUrl &&
+              <NavShareIcon
+                endingText=' – shared using Podverse'
```
[the remark this answers is not in the document]
Comment (mitchdowney) [reply src/screens/PodcastScreen.tsx]: Good idea but the NavShareIcon logic is a bit weird, and I'd rather just leave it for now since we might want the endingText to stay flexible.

File: src/screens/PodcastScreen.tsx [hunk 3]
```diff
@@ -545,9 +573,19 @@ export class PodcastScreen extends React.Component<Props, State> {
...
+    let isSubscribed = subscribedPodcastIds.some(
       (x: string) => x === podcastId
     )
+
+    if (!isSubscribed) {
+      const subscribedPodcasts = safelyUnwrapNestedVariable(
+        () => this.global.subscribedPodcasts,
+        []
+      )
+      isSubscribed = subscribedPodcasts.some((x: any) => x.addByFeedUrl && x.addByFeedUrl === podcastId)
```
[the remark this answers is not in the document]
Comment (mitchdowney) [reply src/screens/PodcastScreen.tsx]: Yeah, in other words it is addByRSSPodcastFeedUrl. That seemed too wordy but I'll go ahead and use it since it's more clear.

File: src/state/actions/downloads.ts [hunk 4]
```diff
@@ -47,19 +47,23 @@ export const initDownloads = async () => {
   const downloadedPodcastEpisodeCounts = await getDownloadedPodcastEpisodeCountsService()
   const downloadedPodcasts = await getDownloadedPodcastsService()
   const autoDownloadSettings = await getAutoDownloadSettingsService()
-  const downloadedEpisodeLimitCount = await AsyncStorage.getItem(PV.Keys.DOWNLOADED_EPISODE_LIMIT_GLOBAL_COUNT)
-  const downloadedEpisodeLimitDefault = await AsyncStorage.getItem(PV.Keys.DOWNLOADED_EPISODE_LIMIT_GLOBAL_DEFAULT)
+  const downloadedEpisodeLimitCount = AsyncStorage.getItem(PV.Keys.DOWNLOADED_EPISODE_LIMIT_GLOBAL_COUNT)
+  const downloadedEpisodeLimitDefault = AsyncStorage.getItem(PV.Keys.DOWNLOADED_EPISODE_LIMIT_GLOBAL_DEFAULT)
```
[the remark this answers is not in the document]
Comment (mitchdowney) [reply src/state/actions/downloads.ts]: Oh good catch. I just forgot the awaits.

File: src/state/actions/downloads.ts (hunk 1 above)
Comment (mitchdowney) [reply src/state/actions/downloads.ts]: Something weird was happening where the state would load correctly on the home screen, with all the subscribed podcasts, but then quickly disappear and be replaced with an empty global state. initDownloads is only called in one place on app launch. Maybe there is a timing issue with getAuthUserInfo or initPlayerState initializeScreenData?

Comment (mitchdowney) [reply src/state/actions/downloads.ts]: @kreonjr when I remove the setTimeout, I am unable to reproduce locally...I spent a few hours trying to fix this issue, and I was having some nasty cache issues that I couldn't seem to fix. Anyway, I'm not sure this race condition even actually exists. I'd say it doesn't hurt to use a setTimeout in initDownloads though so I'm inclined to leave it here.
